src/IndexService/Worker/Helper/MySql.php

```diff
@@ -43,21 +43,37 @@ public function __construct(MysqlConfigInterface $tenantConfig, Connection $db)
     /**
      * @return string[]
      */
-    public function getValidTableColumns(string $table): array
+    public function getPrimaryKey(string $table, bool $cache = true): array
+    {
+        return $this->getValidTableColumns($table, $cache, true);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getValidTableColumns(string $table, bool $cache = true, bool $primaryKeyColumnsOnly = false): array
     {
         $cacheKey = 'plugin_ecommerce_productindex_columns_' . $table;
 
         if (!Cache\RuntimeCache::isRegistered($cacheKey)) {
```

Comment:
`$cache` doesn't seem in use, was meant?
```suggestion
        if (!Cache\RuntimeCache::isRegistered($cacheKey) || !$cache) {
```

but i don't get the changes in the core 
https://github.com/pimcore/pimcore/blob/230610923ee36ab370261fd13a84619be0162ad8/lib/Model/Dao/AbstractDao.php#L72
even though this class doens't extends that, doesn't need to have the same signature